framework/include/cppmicroservices/detail/BundleTrackerPrivate.h

```diff
@@ -0,0 +1,165 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+
+#ifndef CPPMICROSERVICES_BUNDLETRACKERPRIVATE_H
+#define CPPMICROSERVICES_BUNDLETRACKERPRIVATE_H
+
+#include "cppmicroservices/Bundle.h"
+#include "cppmicroservices/BundleContext.h"
+#include "cppmicroservices/Constants.h"
+#include "cppmicroservices/detail/Threads.h"
+#include "cppmicroservices/detail/TrackedBundle.h"
+
+#include <memory>
+#include <stdexcept>
+#include <utility>
+
+namespace cppmicroservices {
+
+template<class T>
+class BundleTracker;
+
+namespace detail {
+
+/**
+ * \ingroup MicroServices
+ */
+template<class T>
+class BundleTrackerPrivate : MultiThreaded<>
+{
+
+public:
+  using BundleStateMaskType = std::underlying_type_t<Bundle::State>;
+
+  BundleTrackerPrivate(BundleTracker<T>* _bundleTracker,
+                       BundleContext const& _context,
+                       const BundleStateMaskType _stateMask,
+                       const std::shared_ptr<BundleTrackerCustomizer<T>> _customizer)
+      : context(_context)
+      , stateMask(_stateMask)
+      , customizer(_customizer)
+      , listenerToken()
+      , trackedBundle()
+      , bundleTracker(_bundleTracker)
+  {
+    this->customizer = customizer;
```

Comment:
Why do we set the customizer to be itself here after already including the customizer initialization in the initialization list?

Reply:
this is a bug. can you fix it? Also, can you rename all the parameter names, removing the `_` and place an `_` in the member variable names for all names that are the same as the parameter name. also take a look in all other sources files for this naming problem and fix as well. thanks!

Reply:
fixed